deltacat/compute/compactor/model/round_completion_info.py

```diff
@@ -7,32 +7,28 @@
 from deltacat.compute.compactor.model.primary_key_index import \
     PrimaryKeyIndexVersionLocator
 
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, List
 
 
 class RoundCompletionInfo(dict):
     @staticmethod
-    def of(high_watermark: int,
+    def of(high_watermark: Optional[Dict[str, int], int],
```

Comment:
In case of compacting deltas from multi-sources, this dict records the high watermark for each source, e.g.,
```
high_watermark = {"src1":100,"src2":200} # src1 or src2 is the canonical_string of partition_locator
```

Reply:
Interesting generalization, but shouldn't there always just be one source partition locator associated with a compaction job run? If we need this generalization, why not take a `PartitionLocator` instead of a string directly for clarity as the key of the dictionary, and then get the canonical string from the input partition locator?

Reply:
1. Yes, there is always one source. I realized, however, when we need to deal with multiple inputs (e.g., new delta, and existing compacted table), there will be multiple `last_stream_compacted`, i.e., last stream compacted for source, and last stream compacted for compacted table. We could persist both for later debugging purpose
2. This info will be persisted into rcf, not sure if we can use object as key

Reply:
1. When we're compacting from another compacted table, won't this always be captured in `rebase_source_partition_locator` already?
2. `PartitionLocator` will be serialized as a JSON dictionary, so it should be OK to use as the key if necessary (dictionary key equality should still work as well AFAIK - let me know if I'm missing something here though).

Reply:
1. yes, it's true
2. sounds good.
Maybe the reason I had a dict there is to preserve the partition locator info (as key), to make sure this high watermark is from the desired locator. It is not necessary to persist other `high_watermark`.

Reply:
Changed the type from string to PartitionLocator